lib/Tool/Authentication.php

```diff
@@ -57,26 +62,91 @@ public static function authenticateSession(Request $request = null): ?User
             }
         }
 
-        if (!Session::requestHasSessionId($request, true)) {
-            // if no session cookie / ID no authentication possible, we don't need to start a session
+        $session = $request->hasPreviousSession() ? $request->getSession() : null;
+
+        if (null === $session) {
             return null;
         }
 
-        $session = Session::getReadOnly();
-        $user = $session->get('user');
+        $token = $session->get('_security_pimcore_admin');
+        $token = $token ? static::safelyUnserialize($token) : null;
 
-        if ($user instanceof User) {
-            // renew user
-            $user = User::getById($user->getId());
+        if ($token instanceof TokenInterface) {
+            $token = static::refreshUser($token, \Pimcore::getContainer()->get(UserProvider::class));
```

Comment:
This feels a bit odd to me, to be honest. 
Shouldn't we try to the he user from `Symfony\Bundle\SecurityBundle\Security` or `security.token_storage` ?

Reply:
reason for that is, that we can authenticate against the Admin Session from outside the firewall. It was there before, so I kept it.

Reply:
I think this is needed to disable frontend cache (for example) when logged into admin

Reply:
Mhm, I see - and there's no other way to get this information outside the firewall (pattern)?

Reply:
Not that I am aware of.

Reply:
What about trying to use `\Symfony\Component\Security\Http\Firewall\ContextListener::authenticate()` with a dummy request going to an admin route, but containing the real session? 

I'm a bit afraid, that something might change how Symfony deals with tokens and then this would completely break. At least we should give it a try 🤔

Reply:
@brusch I am afraid that this is also not going to work, the ContextListener would set the token into the token storage, therefore overwriting the current token. So if we use that, and the user is logged into another firewall, this check would overwrite the token.

Reply:
Hmm, good point. I guess then it's fine if we start that way and we create a follow-up ticket for checking out alternatives, or maybe discuss them with the guys from Symfony.